internal/icon/xpm.go

```diff
@@ -0,0 +1,125 @@
+package icon
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"strconv"
+	"strings"
+)
+
+func parseXPM(data []byte) image.Image {
+	var colCount, charSize int
+	colors := make(map[string]color.Color)
+	var img *image.NRGBA
+
+	rowStart := 0
+	rowNum := 0
+	for i, b := range data {
+		if b == '\n' {
```

Comment:
This entire section could be unnested by changing this if to if b != '\n' { continue }

Reply:
Nice catch thanks